### PR TITLE
Add ch.protonmail.protonmail-import-export-app

### PR DESCRIPTION
**Pull request by A6GibKm**
Manifest mostly copied from [protonmail-bridge](https://github.com/flathub/ch.protonmail.protonmail-bridge)

- The screenshots urls have to be updated after being merged
- Just like [protonmail-bridge](https://github.com/flathub/ch.protonmail.protonmail-bridge), it won't work at all using libsecret >= 0.20, I have reported this in multiple forms in https://github.com/flatpak/xdg-desktop-portal/issues/507 and https://gitlab.gnome.org/GNOME/libsecret/-/issues/49
- The metainfo was created following https://protonmail.com/import-export
- Just like ProtonMail Bridge, the .deb bundles a GPG3.0 license which allows redistribution, but the source is currently not available. I would really like to build the former from source, but until they vendor their dependencies and allow building with native Qt it is way outside of my knowledge.
- I will send an internal bug request providing the metainfo and another politelly asking for co-maintaining.

### CODE REVIEW

**Comment by A6GibKm** [comment]
bot, build ch.protonmail.protonmail-import-export-app

**Comment by A6GibKm** [comment]
bot, build ch.protonmail.protonmail-import-export-app

File: ch.protonmail.protonmail-import-export-app.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - ar x protonmail-import-export-app.deb
+      - tar -xf data.tar.xz
+      - mv usr/lib/* /app/lib/
+      - install -Dp -m 644 usr/share/icons/hicolor/scalable/apps/protonmail-import-export-app.svg /app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -Dp -m 644 usr/share/applications/protonmail-import-export-app.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - desktop-file-edit --set-key="Icon" --set-value=${FLATPAK_ID} /app/share/applications/${FLATPAK_ID}.desktop
+      - desktop-file-edit --set-key="Categories" --set-value="Office;Email" /app/share/applications/${FLATPAK_ID}.desktop
+      - ln -s /app/lib/protonmail/import-export/protonmail-import-export-app /app/bin/
+    sources:
+      - type: file
```
Comment (barthalion) [inline ch.protonmail.protonmail-import-export-app.yml:28]: If there's a deb repo, you should add flatpak-external-data-checker metadata here.

File: libsecret.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (barthalion) [inline libsecret.json:1]: Any reason to store it in a separate file?

**Review by barthalion — COMMENTED** [review]
Did you try to contact Protonmail about this submission?

**Comment by barthalion** [comment]
> Did you try to contact Protonmail about this submission?

Never mind, just read the description.

**Comment by A6GibKm** [comment]
Yes. I contacted them.

File: libsecret.json (hunk 2 above)
Comment (A6GibKm) [reply libsecret.json:1]: Not really, just so it is more in line with the old `- shared-modules/libsecret/libsecret.json`

File: ch.protonmail.protonmail-import-export-app.yml (hunk 1 above)
Comment (A6GibKm) [reply ch.protonmail.protonmail-import-export-app.yml:28]: It is not a repo. But added metadata anyways.

**Comment by A6GibKm** [comment]
They are not interested in co-maintaining for the moment.

**Comment by A6GibKm** [comment]
bot, build ch.protonmail.protonmail-import-export-app

**Pull request closed without merging**